File: app/client/Renty-client/lib/api/dto/request/product/product.dart
import 'package:client/api/dto/request/product/image_create.dart';
import 'package:client/api/dto/response/product/category.dart';
import 'package:client/api/dto/response/product/image.dart';

class ProductCreateRequest {
  final String name;
  final String description;
  final int price;
  final int categoryId;
  final List<ImageCreateRequest> images;

  ProductCreateRequest(
      {required this.name,
      required this.description,
      required this.price,
      required this.categoryId,
      required this.images});

  Map<String, dynamic> toJson() => {
        'name': name,
        'description': description,
        'price': price,
        'category_id': categoryId,
        'images': images,
      };
}

File: app/server/src/main/java/vsu/csf/rentyserver/model/entity/Product.java
package vsu.csf.rentyserver.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.List;

@Entity
@Getter
@Setter
@Accessors(chain = true)
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY,
            generator = "product_sequence")
    @Column(name = "product_id")
    private Long productId;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "price")
    private Integer price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(mappedBy = "sizeId.product", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Size> sizes;

    @OneToMany(mappedBy = "imageId.product", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Image> images;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<RentEvent> rents;

}
